src/cryptonote_core/service_node_list.cpp

```diff
@@ -158,18 +178,47 @@ namespace service_nodes
     }
 
     std::lock_guard<boost::recursive_mutex> lock(m_sn_mutex);
-    auto it = m_quorum_states.find(height);
-    if (it == m_quorum_states.end()) {
-      if (!m_store_quorum_history || !include_old)
-        return nullptr;
-      it = m_old_quorum_states.find(height);
-      if (it == m_old_quorum_states.end())
-        return nullptr;
+    quorum_manager const *quorums = nullptr;
+    if (height == m_state.height)
+      quorums = &m_state.quorums;
+
+    if (!quorums) // NOTE: Search m_state_history
+    {
+      auto it = std::lower_bound(m_state_history.begin(),
+                                 m_state_history.end(),
+                                 height,
+                                 [](state_t const &state, uint64_t height) { return state.height < height; });
+
+      if (it != m_state_history.end())
+      {
+        if (it->height == height)
```

Comment:
Can collapse these two if's into a conjunction: `if (it != m_state_history.end() && it->height == height)`